src/newcharacter.cpp

```diff
@@ -449,11 +450,23 @@ bool player::create(character_type type, std::string tempname)
         break;
     }
 
+    auto nameExists = [&]( const std::string &name ) {
```

Comment:
You can do this without accessing the file system (and without revealing internals of the world factory) by checking `WORLD::save_exists`, and using `save_t::from_player_name` to get a `save_t` instance.

Reply:
Fixed.